packages/web/src/common/store/player/sagas.ts

```diff
@@ -399,7 +434,51 @@ function* recordListenWorker() {
       if (trackId) yield* put(recordListen(trackId))
       lastSeenPlayCounter = playCounter
     }
-    yield* delay(RECORD_LISTEN_INTERVAL)
+
+    if (position) yield* delay(RECORD_LISTEN_INTERVAL)
+  }
+}
+
+const getPlayerSeekInfo = async (audioPlayer: AudioPlayer) => {
+  const position = await audioPlayer.getPosition()
+  const duration = await audioPlayer.getDuration()
+  return {
+    position,
+    duration
+  }
+}
+
+/**
+ * Poll for tracking the playback position of tracks
+ */
+function* recordPlaybackPositionWorker() {
+  const audioPlayer = yield* getContext('audioPlayer')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const isNewPodcastControlsEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.PODCAST_CONTROL_UPDATES_ENABLED
+  )
+
+  // eslint-disable-next-line no-unmodified-loop-condition
+  while (isNewPodcastControlsEnabled) {
+    const trackId = yield* select(getTrackId)
+    const track = yield* select(getTrack, { id: trackId })
+    const playing = yield* select(getPlaying)
+
+    if (trackId && track?.genre === Genre.PODCASTS && playing) {
+      const { position, duration } = yield* call(getPlayerSeekInfo, audioPlayer)
+      const isComplete = PLAYBACK_COMPLETE_SECONDS > duration - position
+      yield* put(
+        setTrackPosition({
+          trackId,
+          positionInfo: {
+            status: isComplete ? 'COMPLETED' : 'IN_PROGRESS',
+            playbackPosition: position
+          }
+        })
+      )
+    }
+    yield* delay(RECORD_PLAYBACK_POSITION_INTERVAL)
```

Comment:
so does this loop even when tracks aren't playing technically? i think action channels might be the best option here, so we only need to run this while a track is actually playing? lmk what you think and maybe us + @amendelsohn can think of what that type of saga would look like.